app/views/plant_family.py

```diff
@@ -0,0 +1,122 @@
+from flask import (
+    Blueprint,
+    render_template,
+    request,
+    flash,
+    redirect,
+    url_for,
+)
+from flask_login import login_required
+import sqlalchemy as sa
+from app.controllers import create_pagination
+
+from app import models as m, db
+from app import forms as f
+from app.logger import log
+
+
+bp = Blueprint("plant_family", __name__, url_prefix="/plant-family")
+
+
+@bp.route("/", methods=["GET"])
+@login_required
+def get_all():
+    q = request.args.get("q", type=str, default=None)
+    where = sa.and_(m.PlantFamily.is_deleted.is_(False))
+
+    if q:
+        where = sa.and_(m.PlantFamily.name.ilike(f"%{q}%"), m.PlantFamily.is_deleted.is_(False))
+
+    query = m.PlantFamily.select().where(where).order_by(m.PlantFamily.id.desc())
+    count_query = sa.select(sa.func.count()).where(where).select_from(m.PlantFamily)
+    pagination = create_pagination(total=db.session.scalar(count_query))
+
+    return render_template(
+        "plant_family/plant_families.html",
+        plant_families=db.session.execute(
+            query.offset((pagination.page - 1) * pagination.per_page).limit(pagination.per_page)
+        ).scalars(),
+        page=pagination,
+        search_query=q,
+    )
+
+
+@bp.route("/create", methods=["GET", "POST"])
+@login_required
+def create():
+    form = f.PlantFamilyForm()
+    form.pests.choices = db.session.scalars(sa.Select(m.Pest.name).where(m.Pest.is_deleted.is_(False))).all()
+    form.illnesses.choices = db.session.scalars(sa.Select(m.Illness.name).where(m.Illness.is_deleted.is_(False))).all()
```

Comment:
I assume we do these queries only when we receive "GET" method? If we receive "POST" request with form these form fields should be pre-occupied, isn't it?